### ISSUE TITLE
Few bugs in the console log example

### ISSUE THREAD

**Issue by aybe** [ocornut/imgui#5721]
### Config/Build Information:

```
Dear ImGui 1.87 (18700)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: IMGUI_DISABLE_OBSOLETE_FUNCTIONS
define: _WIN32
define: _WIN64
define: _MSC_VER=1930
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: NULL
io.BackendRendererName: NULL
io.ConfigFlags: 0x00000000
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00000008
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,256
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```
### Version/Branch of Dear ImGui:

Version: 1.87
Branch: master

Back-end/Renderer/Compiler/OS

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Compiler: Visual Studio
Operating System: Windows 10

### My Issue/Question:

I'm writing my own console based on your example in the mega demo, I've found few issues on it.

### Problem 1:

Add a bunch of entries so that the scrollbar becomes very small, you cannot scroll anymore because the window resize grip takes precedence:

![TestDI_iHQ8NgWjSV](https://user-images.githubusercontent.com/1537591/192697986-c3ccf237-8d7f-4309-a4bd-27e5fa0e320e.png)

If you shrink it enough so that horizontal scrolling appears, the problem is fixed:

![TestDI_AcBezstnak](https://user-images.githubusercontent.com/1537591/192698130-9f8ab911-acb6-4bb3-8290-e7a783140bd3.png)

To try address this in a consistent way, I added an empty `TextUnformatted` after `EndChild` and in `Begin` add the flag `NoScrollbar`:

![TestDI_43c68B4aKP](https://user-images.githubusercontent.com/1537591/192698812-2edee358-4db9-4365-8168-e2418349fb31.png)

That is already much better even though that now you always have an horizontal scrolling even if there's nothing to scroll.

Any idea on how to fix the problem more elegantly?

### Problem 2:

Resizing the window when auto scroll is enabled will screw the vertical position.

When the scroll is say in the middle, all is good:

![TestDI_YfiIYpFshJ](https://user-images.githubusercontent.com/1537591/192699264-05c7de7a-d9d0-4e81-8a30-46d6d0f3aefc.gif)

But see how when auto scroll is on, the scroll bar loses its position:

![TestDI_Q1xBINT2uK](https://user-images.githubusercontent.com/1537591/192699443-e4d71a8b-5e66-43ea-a76c-775dbcc3a0f8.gif)

I have no idea on how to fix that one out.

Any tips are much appreciated, thank you!

**Comment by ocornut** [comment]
> you cannot scroll anymore because the window resize grip takes precedence:

You can still click slightly above.

> Any idea on how to fix the problem more elegantly?

Your style should use more suitable values for ScrollbarSize, WindowPadding etc.
It looks like your font has been scaled but not your style.

> Resizing the window when auto scroll is enabled will screw the vertical position.

Not sure yet.

**Comment by ocornut** [comment]
Looking at the Scroll and ScrollMax values in Metrics/Debugger
![image](https://user-images.githubusercontent.com/8225057/192753541-636b4936-a7aa-4514-a80c-07c1d2a5a9d1.png)

I realized that when the child is clipped, items are coarse clipped (not submitted) so even though scroll max is preserved, we call `ImGui::SetScrollHereY()` while cursor has not moved to it resets scrolling.

Testing the return of `BeginChild()` is an adequate fix for that. I have pushed a fix: 5c4426c

**Comment by aybe** [comment]
Thank you. 😁
